citar-latex.el

```diff
@@ -65,17 +65,58 @@ the point."
   (ignore-errors (reftex-get-bibfile-list)))
 
 ;;;###autoload
-(defun citar-latex-keys-at-point ()
-  "Return a list of keys at point in a latex buffer."
-  (unless (fboundp 'TeX-current-macro)
-    (error "Please install AUCTeX"))
-  (when (citar-latex-is-a-cite-command (TeX-current-macro))
-    (split-string (thing-at-point 'list t) "," t "[{} ]+")))
+(defun citar-latex-key-at-point ()
+  "Return citation key at point with its bounds.
+The return value is (KEY . BOUNDS), where KEY is the citation key
+at point and BOUNDS is a pair of buffer positions.  Returns nil
+if there is no key at point."
```

Comment:
```suggestion
  "Return citation key at point with its bounds.
  
The return value is (KEY . BOUNDS), where KEY is the citation key
at point and BOUNDS is a pair of buffer positions.  

Return nil if there is no key at point."
```

I'm just suggesting some trivial docstring formatting etc changes.

But substantive questions here:

1. Why did this need to get more complex?
2. How does a user change an existing key?

Reply:
> Why did this function need to get more complex? Just curious.

The previous version had false positives with cases like this:
```latex
\citet[see][p.~99]{smith90}
```
If point was in the affixes, it would consider "see" or "p.~99" to be citation keys. In the new code, a citation key is a comma-delimited item inside a pair of curly braces within the citation macro. I checked that this holds for the Natbib and Biblatex macros. Now, if point is on the affixes, `key-at-point` will return nil and Embark will fall back to acting on the entire macro.

There is still a corner case: in the Biblatex "multi-volume" commands, the volume number also appears within braces. I figured we could deal with this later when we revisit more complicated citation macros.

This code can probably be simplified a little using `syntax-ppss`, similar to the Markdown citation parser, but it wasn't immediately clear to me how to do that.